pq/observe_test.go

```diff
@@ -0,0 +1,392 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package pq
+
+import (
+	"runtime"
+	"testing"
+
+	txfile "github.com/elastic/go-txfile"
+	"github.com/elastic/go-txfile/internal/mint"
+)
+
+type testObserveLast statEntry
+
+type statEntry struct {
+	kind statKind
+	off  uintptr
+
+	// OnInit statis
+	version   uint32
+	available uint
+
+	// per callback stats
+	flush FlushStats
+	read  ReadStats
+	ack   ACKStats
+}
+
+type statKind uint8
+
+const (
+	statNone statKind = iota
```

Comment:
Good to create a statNone that will take care of the default value of uint8. An alternative is to do iota+1.